File: src/test/java/pl/laskowski/marcin/domain/algorithm/BruteForceSolverTest.java
package pl.laskowski.marcin.domain.algorithm;

/**
 * Created by Marcin Laskowski.
 */

public class BruteForceSolverTest {

//    private BruteForceSolver algorithm;
//
//    @Before
//    public void setUp() {
//        SudokuDivider divider =new SingleRegionDivider();
//        algorithm = new BruteForceSolver(divider);
//    }
//
//    @Test
//    public void shouldSolveAlreadySolvedGrid() {
//        Sudoku expectedSolution = new Sudoku(1, 1, new Integer[] {1});
//
//        Sudoku foundSolution = algorithm.solve(expectedSolution);
//
//        assertEquals(expectedSolution, foundSolution);
//    }
//
//    @Test
//    public void shouldSolveEmptyGrid() {
//        Sudoku input = new Sudoku(1, 1, new Integer[]{0});
//        Sudoku expectedSolution = new Sudoku(1, 1, new Integer[] {1});
//
//        Sudoku foundSolution = algorithm.solve(input);
//
//        assertEquals(expectedSolution, foundSolution);
//    }
//
//    @Test
//    public void shouldSolveIncompleteGrid() {
//        Sudoku input = new Sudoku(2, 1, new Integer[] {1, 0});
//        Sudoku expectedSolution = new Sudoku(2, 1, new Integer[] {1, 2});
//
//        Sudoku foundSolution = algorithm.solve(input);
//
//        assertEquals(expectedSolution, foundSolution);
//    }
//
//    @Test
//    public void shouldSolveIncompleteGridAtTheEnd() {
//        Sudoku input = new Sudoku(2, 1, new Integer[] {0, 1});
//        Sudoku expectedSolution = new Sudoku(2, 1, new Integer[] {2, 1});
//
//        Sudoku foundSolution = algorithm.solve(input);
//
//        assertEquals(expectedSolution, foundSolution);
//    }
//
//    @Test
//    public void shouldReturnOriginIfThereIsNoSolution() {
//        Sudoku input = new Sudoku(5, 1, new Integer[] {1, 0, 0, 0, 0});
//
//        Sudoku foundSolution = algorithm.solve(input);
//
//        assertEquals(input, foundSolution);
//    }
//
//    @Test
//    public void shouldReturnToPreviousFields() {
//        SudokuDivider divider = new ComplexDivider(
//                new RowDivider(),
//                new ColumnDivider()
//        );
//
//        algorithm = new BruteForceSolver(divider, 3);
//
//        Sudoku input = new Sudoku(3, 2, new Integer[] {
//                0, 0, 0,
//                0, 0, 3
//        });
//        Sudoku expectedSolution = new Sudoku(3, 2, new Integer[] {
//                1, 3, 2,
//                2, 1, 3
//        });
//
//        Sudoku foundSolution = algorithm.solve(input);
//
//        assertEquals(expectedSolution, foundSolution);
//    }

}

File: app/src/test/java/pl/laskowski/marcin/domain/algorithm/BruteForceSolverTest.kt
package pl.laskowski.marcin.domain.algorithm

import org.junit.Assert.*
import org.junit.Test
import pl.laskowski.marcin.model.Sudoku
import pl.laskowski.marcin.model.type.ClassicSquare1x1
import pl.laskowski.marcin.model.type.ClassicSquare2x2
import pl.laskowski.marcin.solving.bruteforce.BruteForceSolver

class BruteForceSolverTest {

    private val algorithm = BruteForceSolver()

    @Test
    fun shouldSolveAlreadySolvedGrid() {
        val expectedSolution = Sudoku(ClassicSquare1x1, listOf(1))
        val foundSolution = algorithm.solve(expectedSolution)
        assertEquals(expectedSolution, foundSolution)
    }

    @Test
    fun shouldSolveEmptyGrid() {
        val input = Sudoku(ClassicSquare1x1, listOf(0))
        val expectedSolution = Sudoku(ClassicSquare1x1, listOf(1))
        val foundSolution = algorithm.solve(input)
        assertEquals(expectedSolution, foundSolution)
    }

    @Test
    fun shouldSolveIncompleteGrid() {
        val input = Sudoku(ClassicSquare2x2, listOf(1, 2, 0, 0))
        val expectedSolution = Sudoku(ClassicSquare2x2, listOf(1, 2, 2, 1))
        val foundSolution = algorithm.solve(input)
        assertEquals(expectedSolution, foundSolution)
    }

    @Test
    fun shouldSolveIncompleteGridAtTheEnd() {
        val input = Sudoku(ClassicSquare2x2, listOf(0, 0, 1, 2))
        val expectedSolution = Sudoku(ClassicSquare2x2, listOf(2, 1, 1, 2))
        val foundSolution = algorithm.solve(input)
        assertEquals(expectedSolution, foundSolution)
    }

    @Test
    fun shouldReturnOriginIfThereIsNoSolution() {
        val input = Sudoku(ClassicSquare2x2, listOf(1, 1, 0, 0))
        val foundSolution = algorithm.solve(input)
        assertEquals(input, foundSolution)
    }



//    @Test
//    fun shouldReturnToPreviousFields() {
//        val input = Sudoku(
//            3, 2, arrayOf(
//                0, 0, 0,
//                0, 0, 3
//            )
//        )
//        val expectedSolution = Sudoku(
//            3, 2, arrayOf(
//                1, 3, 2,
//                2, 1, 3
//            )
//        )
//        val foundSolution = algorithm.solve(input)
//        assertEquals(expectedSolution, foundSolution)
//    }
}